plugins/source/azure/resources/services/monitor/diagnostic_settings.go

```diff
@@ -1,281 +1,130 @@
+// Auto generated code - DO NOT EDIT.
+
 package monitor
 
 import (
 	"context"
-	"errors"
 
-	"github.com/Azure/azure-sdk-for-go/services/preview/monitor/mgmt/2021-07-01-preview/insights"
-	"github.com/Azure/go-autorest/autorest/azure"
 	"github.com/cloudquery/cloudquery/plugins/source/azure/client"
-	"github.com/cloudquery/cq-provider-sdk/provider/diag"
 	"github.com/cloudquery/cq-provider-sdk/provider/schema"
-	"golang.org/x/sync/errgroup"
-)
-
-// diagnosticSetting is a custom copy of insights.DiagnosticSettingsResource with extra ResourceURI field
-type diagnosticSetting struct {
```

Comment:
There's a bit of redundancy here. We only need to add `ResourceURI` as `ID,Name,Type` are on the embedded `DiagnosticSettings` struct so this becomes:
```go
type diagnosticSettingResource struct {
	insights.DiagnosticSettingsResource
	ResourceURI string
}
```